# origin: Debian (Jochen Voss)
# cset: n/a
# inclusion: not submitted
# description: don't panic on bad mptable signature
# revision date: 2004-09-04

diff -urN kernel-source-2.4.26/arch/i386/kernel/mpparse.c kernel-source-2.4.26-1/arch/i386/kernel/mpparse.c
--- kernel-source-2.4.26/arch/i386/kernel/mpparse.c	2004-04-14 23:05:25.000000000 +1000
+++ kernel-source-2.4.26-1/arch/i386/kernel/mpparse.c	2004-04-17 14:23:36.000000000 +1000
@@ -429,15 +429,12 @@
 	unsigned char *bus_data;
 
 	if (memcmp(mpc->mpc_signature,MPC_SIGNATURE,4)) {
-		panic("SMP mptable: bad signature [%c%c%c%c]!\n",
-			mpc->mpc_signature[0],
-			mpc->mpc_signature[1],
-			mpc->mpc_signature[2],
-			mpc->mpc_signature[3]);
+		printk(KERN_ERR "SMP mptable: bad signature [0x%x]!\n",
+			*(u32 *)mpc->mpc_signature);
 		return 0;
 	}
 	if (mpf_checksum((unsigned char *)mpc,mpc->mpc_length)) {
-		panic("SMP mptable: checksum error!\n");
+		printk(KERN_ERR "SMP mptable: checksum error!\n");
 		return 0;
 	}
 	if (mpc->mpc_spec!=0x01 && mpc->mpc_spec!=0x04) {

